qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_model.py

```diff
@@ -236,7 +236,10 @@ def update_log_workspace_group(self):
         # update log tables
         self.remove_all_log_rows()
         for irow, (ws_name, ws) in enumerate(self._data_workspaces.get_loaded_ws_dict().items()):
-            self.add_log_to_table(ws_name, ws, irow)
+            try:
+                self.add_log_to_table(ws_name, ws, irow)
+            except Exception as e:
```

Comment:
The exception I was seeing was from these lines in `add_log_to_table`:
```
        row = [ws.getInstrument().getFullName(), ws.getRunNumber(), run.getProperty('bankid').value,
               run.getProtonCharge(), ws.getTitle()]
        self.write_table_row(ADS.retrieve("run_info"), row, irow)
```
The bankid on some old .nxs files was a number rather than a string. I've fixed this but I figured it was useful to still handle any other exceptions. The UI gets itself in a bit of a tangle if you bomb out here because the dictionaries in the model have been updated but the data doesn't show up in the data widget table so things get out of sync.

Reply:
Yep good point that if it falls over in `add_log_to_table ` it can get the model in a state - putting the try/except where you have is definitely the safest thing.